packages/twenty-server/src/engine/core-modules/record-crud/services/create-record.service.ts

```diff
@@ -83,14 +86,61 @@ export class CreateRecordService {
           objectMetadataMapItem: objectMetadataItemWithFieldsMaps,
         });
 
-      const insertResult = await repository.insert({
+      let selectedColumns = undefined;
+      let dataToInsert = {
         ...transformedObjectRecord,
         position,
         createdBy: params.createdBy ?? {
           source: FieldActorSource.WORKFLOW,
           name: 'Workflow',
         },
-      });
+      };
+
+      const restrictedFields =
+        repository.objectRecordsPermissions?.[
+          objectMetadataItemWithFieldsMaps.id
+        ]?.restrictedFields;
+
+      if (!isEmpty(restrictedFields)) {
+        const selectableFields = getAllSelectableFields({
+          restrictedFields,
+          objectMetadata: {
+            objectMetadataMapItem: objectMetadataItemWithFieldsMaps,
+          },
+        });
+
+        selectedColumns = Object.keys(selectableFields).filter(
+          (key) => selectableFields[key],
+        );
+
+        const updatableFields = getAllUpdatableFields({
+          restrictedFields,
+          objectMetadata: {
+            objectMetadataMapItem: objectMetadataItemWithFieldsMaps,
+          },
+        });
+
+        const filteredData = Object.fromEntries(
+          Object.entries(transformedObjectRecord).filter(
+            ([key]) => updatableFields[key] !== false,
+          ),
+        );
+
+        dataToInsert = {
```

Comment:
@FelixMalfait @abdulrahmancodes discussed this @ijreilly, this looks fine but from a product perspective, I wonder if we should rather let the record creation fail if the agent / workflow node is trying to update a field it cannot update (the agent could retry with a smaller set of field but would be aware). Here it will think that it succeeds to update all fields but it will actually only update the subset it has permission to